### PR TITLE
ipc4: dai: remove S32_LE hardcode for SSP DAIs

### PR DESCRIPTION
**Pull request by kv2019i**
The limitation that SSP sample format is always S32_LE does not hold. There can be I2S codecs where 16bit sample width is used and SSP FIFO will expect 16bit samples from DMA.

Signed-off-by: Kai Vehmanen <kai.vehmanen@linux.intel.com>

### CODE REVIEW

**Review by ranj063 — APPROVED** [review]

**Review by juimonen — APPROVED** [review]

**Comment by kv2019i** [comment]
Tests seem good, but marking as DNM still for a while. Still a few unknowns w.r.t. format propagation I want to double-check before we proceed to merge this.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@kv2019i needed for mtl-03 ? @mwasko fyi.

**Comment by kv2019i** [comment]
Format propagation issue clearer and I see no reason to hold this PR anymore.

**Comment by kv2019i** [comment]
SOFCI TEST

**Pull request merged**
